### DISCUSSION TITLE
[New Feature] Unlock self-healing cap

### DISCUSSION THREAD

**Discussion by MikillRosen** [Vinifera-Developers/Vinifera#1450] [Feature Requests]
### Description:
The self-healing ability usually stops at 50% health. This feature removes this limit.

### Possible Implementation:
`SelfHealing.Cap=` is compatible to the [General] section and TechnoTypes in rules.ini and firestrm.ini. It changes how much health it can self heal before stopping. It accepts the values between `0%` and `100%`. Defaults to `50%`.

**Comment by AtifArshad1995** [comment]
To be a bit more specific, it doesn't stop at 50% per se. It stops when the health bar turns green and the green status happens to be set at 50% health in vanilla TS as shown by lines in rules.ini

ConditionRed=25%        ; when damaged to this percentage, health bar turns red
**ConditionYellow=50%     ; when damaged to this percentage, health bar turns yellow**

If we increase ConditionYellow to say 66%, units self heal to 66% and when health turns green, it stops.

**Comment by JoyfulShush** [comment]
I would love allowing units to fully self-heal, that sounds like a design that can work well for some setups.

That said, I think it should not be made globally, but instead, should be done on units directly. This way, you can make an epic superunit fully heal, while a harvester would still only heal to green health (50% in vanilla TS).

Additionally, the Elite ability that enables self heal should respect that unit's self heal cap, if it has one, or default to just being green by default like it is today.

**Reply by MikillRosen** [reply]
if you don't want it to be global, then don't edit the global value. I suggested the global value so you can easily set a selfheal cap on all units without doing it individually. Adding the flag on individual units would still supersede the global value.

**Reply by JoyfulShush** [reply]
Fair enough, using the global value as a default/fallback is a great idea.
